ts/src/header-validator/validate-json.ts

```diff
@@ -907,6 +1003,30 @@ export type TriggerSpec = {
   triggerData: Set<number>
 }
 
+export type SourceAggregatableDebugReportingConfig =
+  AggregatableDebugReportingConfig & {
+    budget: number
+  }
+
+function sourceAggregatableDebugReportingConfig(
+  ctx: RegistrationContext,
+  j: Json
+): Maybe<SourceAggregatableDebugReportingConfig> {
+  return struct(ctx, j, {
+    budget: field('budget', aggregatableValue),
+
+    ...aggregatableDebugReportingConfig,
+  }).filter((s) => {
+    for (const d of s.debugData) {
+      if (d.value > s.budget) {
+        ctx.error(`data contains value greater than budget (${s.budget})`)
```

Comment:
Would it be possible to use the scope corresponding to the array item's value field?

One way to do that would be to process the `budget` field first, and pass it in to the function that validates the debug data.

Reply:
Alternatively, I can process `budget` as suggested, and make that non-StructFields, similar to the aggregation coordinator origin. Wdyt?

Reply:
We do something similar in https://github.com/WICG/attribution-reporting-api/blob/d6bbe748163187f2c017a6940f5dc024dc0b6169/ts/src/header-validator/validate-json.ts#L473-L492 where we process one field before the other.

Reply:
yeah, but then it can't be a `StructFields` IIUC?

Reply:
Feel free to leave it for now. I'll come in and see what can be added later. Maybe just add a TODO for giving more precise path information.

Reply:
Added TODO, thanks.